dask_sql/physical/rex/core/call.py

```diff
@@ -688,57 +688,32 @@ def timestampadd(self, unit, interval, df: SeriesOrScalar):
             raise RuntimeError(f"Negative time interval {interval} is not supported.")
         df = df.astype("datetime64[ns]")
 
+        if "cudf" in str(type(df)):
+            from cudf import DateOffset
+        else:
+            from pandas.tseries.offsets import DateOffset
```

Comment:
Not directly related to this pr but I was thinking we could have a util within dask-sql or use an exisiting dask-df util to check this instead of having similar yet slightly different checks in our codebase.

Reply:
A followup issue for this would be good as well